src/pages/index.js

```diff
@@ -143,16 +158,48 @@ const editProfilePopup = new PopupWithForm(
 );
 editProfilePopup.setEventListeners();
 
+const avatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
 // ---------------- Form Handlers ----------------
 function handleProfileFormSubmit(values) {
   console.log(values);
-  userInfo.setUserInfo({
-    title: values.title,
-    description: values.description,
-  });
-  /* closeModal(profileEditModal); */
-  // closeModal(editProfilePopup);
-  editProfilePopup.close();
+  renderLoading("#profile-edit-modal", true);
+  api
+    .updateUserInfo({
+      name: values.title,
+      about: values.description,
+    })
+    .then((data) => {
+      console.log(data);
+      userInfo.setUserInfo(data);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error updating user info: ${err}`);
+    })
+    .finally(() => {
+      renderLoading("#profile-edit-modal", false);
+    });
+}
+
+function handleAvatarFormSubmit(values) {
+  renderLoading("#avatar-edit-modal", true);
+  api
+    .updateUserAvatar(values.avatar)
+    .then((data) => {
+      userInfo.setUserAvatar(data.avatar);
+      avatarPopup.close();
+      // renderLoading("#avatar-edit-modal", false);
+    })
+    .catch((err) => {
+      console.error(`Error updating avatar: ${err}`);
+      // renderLoading("#avatar-edit-modal", false);
+    })
+    .finally(renderLoading("#avatar-edit-modal", false));
```

Comment:
`    .finally(renderLoading("#avatar-edit-modal", false));
`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js
    .finally(() => {
      renderLoading("#avatar-edit-modal", false);
    });
```